Cabal/Distribution/Simple/Test/LibV09.hs

```diff
@@ -99,8 +99,15 @@ runTest pkg_descr lbi flags suite = do
                                                    lbi
                                                    (LBI.CTestName
                                                       (PD.testName suite))
+                                      (_, _, _, _, _, _, exeClbi) =
+                                          testSuiteLibV09AsLibAndExe
+                                              pkg_descr suite clbi lbi
+                                              distPref pwd
+                                  -- Get the dependencies of the stub
+                                  -- executable. They should include the test
+                                  -- library itself (#2039).
                                   paths <- LBI.depLibraryPaths
-                                             True False lbi clbi
+                                             True False lbi exeClbi
```

Comment:
I investigated and here is what I discovered:

1. Here is something confusing: the function is called `depLibraryPaths` (library?!), yet we are passing it the CLBI for the executable! I looked at the definition and the point is that depLibraryPaths doesn't have anything to do about libraries; it has to do with the DEPENDENT libraries, and whose dependencies do we care about? The executables. So I guess I'd personally be a fan of a ``-- NB: the library path is for the dependencies of the *executable*, not library`` (or if you want to shave yaks, renaming `depLibraryPaths`)
2. I don't understand why this code has to go through all these gyrations to even get its hands on the executable CLBI. This seems wrong-headed. The real problem is that the CLBIs for this type of test is a massive hack, as explained in Build:

```
               clbi -- This ComponentLocalBuildInfo corresponds to a detailed
                    -- test suite and not a real component. It should not
                    -- be used, except to construct the CLBIs for the
                    -- library and stub executable that will actually be
                    -- built.
```

   A variant of this comment seems apt here. If this were GHC I'd tell you to refer to the Note.

Does that help?